08_cell_line_prediction/nn_diagnostics/nbconverted/nn_hsize_range.py

```diff
@@ -0,0 +1,191 @@
+sns.set({'figure.figsize': (24, 10)})
+
+hsize_vals = sorted(list([int(k) for k in hsize_dfs.keys()]))
+print(hsize_vals)
+
+fig, axarr = plt.subplots(ceil(len(hsize_vals) / 3), 3)
+
+for ix, hsize in enumerate(hsize_vals):
+    ax = axarr[ix // 3, ix % 3]
+    if ix == 0:
+        sns.lineplot(data=hsize_dfs[str(hsize)], x='epoch', y='value', hue='dataset', ax=ax)
+    else:
+        sns.lineplot(data=hsize_dfs[str(hsize)], x='epoch', y='value', hue='dataset', legend=False, ax=ax)
+    ax.set_xlabel('Epoch')
+    ax.set_ylabel('AUPR')
+    ax.set_ylim(0, 1.1)
+    ax.set_title(f'Learning curve for {plot_gene}, hsize={hsize}, averaged over splits')
+    
+plt.tight_layout()
+
+
+# In[5]:
+
+
+sns.set_style('whitegrid')
+sns.set({'figure.figsize': (24, 10)})
+
+hsize_vals = sorted(list([int(k) for k in hsize_dfs.keys()]))
+print(hsize_vals)
+
+fig, axarr = plt.subplots(ceil(len(hsize_vals) / 3), 3)
+
+for hsize_ix, hsize in enumerate(hsize_vals):
+    ax = axarr[hsize_ix // 3, hsize_ix % 3]
+    hsize_df = hsize_dfs[str(hsize)]
+    for fold_ix, fold in enumerate(hsize_df.fold.unique()):
+        if hsize_ix == 0 and fold_ix == 0:
+            sns.lineplot(data=hsize_df[hsize_df.fold == fold],
+                         x='epoch', y='value', hue='dataset', ax=ax)
+        else:
+            sns.lineplot(data=hsize_df[hsize_df.fold == fold],
+                         x='epoch', y='value', hue='dataset',
+                         legend=False, ax=ax)
+    ax.set_xlabel('Epoch')
+    ax.set_ylabel('AUPR')
+    ax.set_ylim(0, 1.1)
+    ax.set_title(f'Learning curve for {plot_gene}, hsize={hsize}, for each split')
+
+plt.tight_layout()
+
+
+# In[6]:
+
+
+hsize_df = pd.concat(hsize_dfs.values())
+print(hsize_df.shape)
+hsize_df.head()
+
+
+# In[7]:
+
+
+last_epoch = hsize_df.epoch.max()
+perf_df = (hsize_df[hsize_df.epoch == last_epoch]
+    .drop(columns=['epoch'])
+)
+                      
+print(perf_df.shape)
+perf_df.head()
+
+
+# In[8]:
+
+
+# same plot as before but with the "best"/"smallest" parameters marked
+sns.set({'figure.figsize': (10, 8)})
+sns.set_style('ticks')
+
+plot_df = (perf_df
+    .sort_values(by=['hsize'])
+    .reset_index(drop=True)
+)
+plot_df.hsize = plot_df.hsize.astype(int)
+
+with sns.plotting_context('notebook', font_scale=1.6):
+    g = sns.pointplot(
+        data=plot_df,
+        x='hsize', y='value', hue='dataset',
+        hue_order=['train', 'cv', 'test'],
+        marker='o'
+    )
+    g.set_xlabel(f'Hidden layer size (lower = more regularization)')
+    g.set_ylabel('AUPR')
+        
+    ax = plt.gca()
+    legend_handles, _ = ax.get_legend_handles_labels()
+    dataset_labels = ['TCGA (train)', 'TCGA (holdout)', 'CCLE'] 
+    ax.legend(legend_handles, dataset_labels, title='Dataset')
+    sns.move_legend(g, "upper left", bbox_to_anchor=(1.01, 1))
+    plt.title(f'Hidden layer size vs. AUPR, {plot_gene}', y=1.025)
+
+
+# In[9]:
+
+
+# same plot as before but with the "best"/"smallest" parameters marked
```

Comment:
Same thing here, I think the code got copied and changed but the comment wasn't. Looks like it's the same as above but with confidence intervals?